update flatten.py docstring

neonwatty opened the pull request:
Carried over comment from tensor.flatten docstring to to clarify when a view vs copy is instantiated - this has been a [minor point of confusion in forums](https://discuss.pytorch.org/t/what-is-the-difference-of-flatten-and-view-1-in-pytorch/51790/5).  This comment is:

```
    Unlike NumPy’s flatten, which always copies input’s data, this function may return the original object, a view, or copy. 
    If no dimensions are flattened, then the original object input is returned. 
    Otherwise, if input can be viewed as the flattened shape, then that view is returned. 
    Finally, only if the input cannot be viewed as the flattened shape is input’s data copied. 
    See torch.Tensor.view() for details on when a view will be returned.
```

neonwatty committed:
update flatten.py docstring 

carried over comment from tensor.flatten docstring to to clarify when a view vs copy is instantiated

mikaylagawarecki commented on torch/nn/modules/flatten.py:
```diff
@@ -9,6 +9,12 @@
 class Flatten(Module):
     r"""
     Flattens a contiguous range of dims into a tensor. For use with :class:`~nn.Sequential`.
+    
```
Since this paragraph is copied from the docs for `torch.flatten`, perhaps we could add a sentence that links readers to the docs for torch.flatten instead :)

jermwatt replied:
sensible

jermwatt replied:
done!

mikaylagawarecki reviewed (commented):
Thanks @jermwatt, could you please sign the CLA agreement (see https://github.com/pytorch/pytorch/pull/97276#issuecomment-1478486167)

neonwatty committed:
Update flatten.py

Amended prior commit to include link to torch.tensor documentation instead copy & paste from said page.

mikaylagawarecki commented on torch/nn/modules/flatten.py:
```diff
@@ -9,6 +9,8 @@
     r"""
     Flattens a contiguous range of dims into a tensor. For use with :class:`~nn.Sequential`.
+    
+    Inherits from torch.flatten, see https://pytorch.org/docs/stable/generated/torch.flatten.html
```
You should be able to create links in the docs like below

```suggestion
    See :meth:`torch.flatten` for details.
```

If you would like you can also check the rendering of the docs under "Preview Python docs built from this PR in the pytorch-bot comment https://github.com/pytorch/pytorch/pull/97276#issuecomment-1478485983

jermwatt replied:
ah yes of course!

neonwatty committed:
Update torch/nn/modules/flatten.py

Replace raw url link to torch.flatten docs with :meth: link.

Co-authored-by: mikaylagawarecki <35276741+mikaylagawarecki@users.noreply.github.com>

mikaylagawarecki commented:
Looks great, thank you so much! One last thing, to fix the failing lint job running `setup_lint` and `quicklint` from your local pytorch directory and following the corresponding instructions should be sufficient, see [here](https://github.com/pytorch/pytorch/blob/d850c33bfe3f1d1f0040738718cacb04ee449bdc/Makefile#L25-L33)

neonwatty committed:
update flatten.py to address lint issue

removed flake8 triggering emptyline introduced in new nn.flatten comment that was causing lint issue

mikaylagawarecki reviewed (approved):
Thanks!

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge
